src/main/java/org/folio/oaipmh/helpers/records/impl/DcRecordHelper.java

```diff
@@ -0,0 +1,31 @@
+package org.folio.oaipmh.helpers.records.impl;
+
+import java.util.Collection;
+import java.util.List;
+
+import javax.xml.bind.JAXBElement;
+import javax.xml.namespace.QName;
+
+import org.folio.oaipmh.helpers.records.RecordHelper;
+import org.openarchives.oai._2.RecordType;
+import org.openarchives.oai._2_0.oai_dc.Dc;
+import org.purl.dc.elements._1.ElementType;
+
+public class DcRecordHelper implements RecordHelper {
+
+  private static final String NAMESPACE_URL = "http://purl.org/dc/elements/1.1/";
+
+  @Override
+  public void updateRecordCollectionWithSuppressDiscoveryData(Collection<RecordType> records) {
+    records.forEach(record -> {
+      boolean isSuppressedFromDiscovery = record.isSuppressDiscovery();
+      String suppressDiscoveryValue = isSuppressedFromDiscovery ? "discovery suppressed" : "discovery not suppressed";
+      Dc dcRecord = (Dc) record.getMetadata().getAny();
+      List<JAXBElement<ElementType>> list = dcRecord.getTitlesAndCreatorsAndSubjects();
+      QName name = new QName(NAMESPACE_URL, "rights");
+      ElementType elementType = new ElementType().withValue(suppressDiscoveryValue);
```

Comment:
Instead of doing this here, does it make sense to add update the marc -> dc xslt?  See https://github.com/folio-org/mod-oai-pmh/blob/fb1a772a45ec28d8b372c6b4f8fa57866751c988/src/main/resources/xslt/MARC21slim2OAIDC.xsl#L191

Reply:
so I think it's being stored in 999$t as 1/0...  so something like:

```
        <xsl:for-each select="marc:datafield[@tag=999]">
            <dc:rights>
                <xsl:value-of select="marc:subfield[@code='t']"/>
            </dc:rights>
        </xsl:for-each>
```

Reply:
I see now that you don't want to use the 1/0 directly in the DC, so you'll probably need to use something like a choose...

```
        <xsl:for-each select="marc:datafield[@tag=999]">
            <dc:rights>
                <xsl:variable name="suppress" select="marc:subfield[@code='t']" />
                <xsl:choose>
                    <xsl:when test="$suppress > 0">
                        suppressed from discovery
                    </xsl:when>
                    <xsl:otherwise>
                        not suppressed from discovery
                    </xsl:otherwise>
                </xsl:choose>            
            </dc:rights>
        </xsl:for-each>
```

Reply:
@craigmcnally ,thank you for advice, but i prefer to add such field via the code by the reason that such field is optinal, i mean it depeneds on "suppress discovery processing setting". And if i'll do like you suggested then instead of adding the field it will be required to filter(suppress) him from the response each time.

Reply:
I was thinking that the marc would be conditionally updated (based on suppress discovery processing setting) with the suppress discovery field 999$t, and then the style sheet is applied for oai_dc.  

The benefit of this approach is that the dc processing is still very straightforward - take the marc and run it through an xslt, instead of having this combination of java and xslt.